hal/src/photon/concurrent_hal.cpp

```diff
@@ -38,52 +52,23 @@ static_assert(sizeof(uint32_t)==sizeof(void*), "Requires uint32_t to be same siz
  * @param stack_size        The size of the stack to create. The stack is allocated on the heap.
  * @return an error code. 0 if the thread was successfully created.
  */
-os_result_t os_thread_create(os_thread_t* result, const char* name, os_thread_prio_t priority, os_thread_fn_t fun, void* thread_param, size_t stack_size)
+os_result_t os_thread_create(os_thread_t* thread, const char* name, os_thread_prio_t priority, os_thread_fn_t fun, void* thread_param, size_t stack_size)
 {
-    wiced_thread_t* pthread = new wiced_thread_t();
-    wiced_result_t error = wiced_rtos_create_thread(pthread, priority, name, wiced_thread_function_t(fun), stack_size, thread_param );
-    if (error) {
-        delete pthread;
-        *result = NULL;
-    }
-    else {
-        *result = pthread;
-    }
-    return error;
+    *thread = NULL;
+    signed portBASE_TYPE result = xTaskCreate( (pdTASK_CODE)fun, (const signed char*)name, (stack_size/sizeof(portSTACK_TYPE)), thread_param, (unsigned portBASE_TYPE) (configMAX_PRIORITIES-1)-priority, thread);
+    return ( result != (signed portBASE_TYPE) pdPASS );
 }
 
-/**
- *
- * @param result            Receives the created thread handle. Will be set to NULL if the thread cannot be created.
- * @param name              The name of the thread. May be used if the underlying RTOS supports it.
- * @param priority          The thread priority. It's best to stick to a small range of priorities, e.g. +/- 7.
- * @param fun               The function to execute in a separate thread.
- * @param thread_param      The parameter to pass to the thread function.
- * @param stack_size        The size of the stack to create. The stack is allocated on the heap.
- * @param stack             The location of the bottom of the stack. The top of the stack is at location stack + stack_size.
- */
-os_result_t os_thread_create_with_stack(os_thread_t* result, const char* name, os_thread_prio_t priority, os_thread_fn_t fun, void* thread_param, size_t stack_size, void* stack)
-{
-    wiced_thread_t* pthread = new wiced_thread_t();
-    wiced_result_t error = wiced_rtos_create_thread_with_stack(pthread, priority, name, wiced_thread_function_t(fun), stack, stack_size, thread_param);
-    if (error) {
-        delete pthread;
-        *result = NULL;
-    }
-    else {
-        *result = pthread;
-    }
-    return error;
-}
+
 
 /**
  * Determines if the given thread is the one executing.
  * @param   The thread to test.
  * @return {@code true} if the thread given is the one currently executing. {@code false} otherwise.
  */
-bool os_thread_current(os_thread_t* thread)
+bool os_thread_current(os_thread_t thread)
```

Comment:
This function is called os_thread_is_current in concurrent_hal.h
It should be renamed in this file.

Adding os_thread_t os_thread_current() to return the current thread would be useful.